### PR TITLE
Appwizard

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: backend/src/yeomanui.ts [hunk 1]
```diff
@@ -221,7 +187,8 @@ export class YeomanUI {
 				logger: this.logger.getChildLogger({ label: generatorName }),
 				vscode: this.getVscode(), // TODO: remove this temporary workaround once a better solution is found,
 				data: this.uiOptions.data,
-				swaTracker: SWA.getSWATracker()
+				swaTracker: SWA.getSWATracker(),
```
Comment (tomer-epstein) [inline backend/src/yeomanui.ts:190]: maybe not in this change, but later we should consider move vscode & swaTracker in appWizard.

File: frontend/src/App.vue [hunk 2]
```diff
@@ -258,11 +259,11 @@ export default {
       this.toShowPromptMessage = true;
       this.promptMessageIcon = image;
 
-      if (type === "error") {
+      if (type === 0) {
```
Comment (tomer-epstein) [inline frontend/src/App.vue:262]: lets make sure the type remain string "Error" or "error"

File: generator-foodq/generators/app/index.js [hunk 3]
```diff
@@ -353,9 +354,8 @@ module.exports = class extends Generator {
 	}
 
 	writing() {
-		if (_.get(this.log, "showProgress")) {
-			this.log.showProgress("FoodQ is generating.");
-		}
+		this.appWizard.messages.showProgress("FoodQ is generating.");
```
Comment (tomer-epstein) [inline generator-foodq/generators/app/index.js:357]: wouldn't it be more clear --> this.appWizard.showProgress("FoodQ is generating.");

File: generator-foodq/generators/app/index.js [hunk 4]
```diff
@@ -400,6 +400,6 @@ module.exports = class extends Generator {
 	}
 
 	end() {
-		this.log('FoodQ ended', { type: "info", location: "message" });
+		this.appWizard.messages.show('FoodQ ended');
```
Comment (tomer-epstein) [inline generator-foodq/generators/app/index.js:403]: wouldn't it be more clear  --> this.appWizard..showMessage('FoodQ ended');

File: types/src/AppWizard.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+export namespace Message {
+	export enum Location {
+		prompt,
+		notification
+	}
+
+	export enum Type {
+		error, warn, info
+	}
+}
```
Comment (tomer-epstein) [inline types/src/AppWizard.ts:45]: what about -->
not exporting class Message {
class AppWizard will expose the funcs showProgress , showMessage

File: backend/src/yeomanui.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply backend/src/yeomanui.ts:190]: vscode should be removed at all (in the future)
not sure about swaTracker

File: generator-foodq/generators/app/index.js (hunk 3 above)
Comment (slavik-lvovsky) [reply generator-foodq/generators/app/index.js:357]: yes, but in the future you will have commands and other objects
now it looks similar to vscode convention

File: generator-foodq/generators/app/index.js (hunk 4 above)
Comment (slavik-lvovsky) [reply generator-foodq/generators/app/index.js:403]: yes, but in the future you will have commands and other objects
now it looks similar to vscode convention

File: types/src/AppWizard.ts (hunk 5 above)
Comment (slavik-lvovsky) [reply types/src/AppWizard.ts:45]: not sure, let's talk about it

File: generator-foodq/generators/app/index.js (hunk 3 above)
Comment (tomer-epstein) [reply generator-foodq/generators/app/index.js:357]: ok

File: generator-foodq/generators/app/index.js (hunk 4 above)
Comment (tomer-epstein) [reply generator-foodq/generators/app/index.js:403]: ok

File: frontend/src/App.vue (hunk 2 above)
Comment (slavik-lvovsky) [reply frontend/src/App.vue:262]: After we publish the new types version I will change it

**Pull request closed without merging**